notebooks/0110_CAMS.py

```diff
@@ -0,0 +1,310 @@
+
+# add missing columns
+country_emissions_df["scenario"] = "historical"  # this is debatable: CAMS has some extrapolation
+world_emissions_df["scenario"] = "historical"
+
+# now pivot to have years as columns, as required
+country_emissions_df = country_emissions_df.pivot(
+    columns="time", index=("variable", "gas", "iso", "scenario"), values="emissions"
+)
+world_emissions_df = world_emissions_df.pivot(columns="time", index=("variable", "gas", "scenario"), values="emissions")
+
+# add units
+units = pd.MultiIndex.from_tuples(
+    [
+        ("BC", "Mt BC/yr"),
+        ("CH4", "Mt CH4/yr"),
+        ("CO", "Mt CO/yr"),
+        ("NH3", "Mt NH3/yr"),
+        ("NMVOC", "Mt NMVOC/yr"),
+        ("NOx", "Mt NO/yr"),
+        ("OC", "Mt OC/year"),
+        ("SO2", "Mt SO2/year"),
+        ("N2O", "Mt N2O/year"),
+        ("CO2", "Mt CO2/year"),
+        ("CO2_excl_short-cycle_org_C", "Mt CO2/year"),
+        ("CO2_short-cycle_org_C", "Mt CO2/year"),
+    ],
+    names=["gas", "unit"],
+)
+
+country_emissions_df = country_emissions_df.rename_axis(index={"iso": "country", "variable": "sector"}).pix.semijoin(
+    units, how="left"
+)
+world_emissions_df = world_emissions_df.pix.semijoin(units, how="left").rename_axis(index={"variable": "sector"})
+
+# add models
+country_emissions_df = country_emissions_df.pix.semijoin(models, how="left")
+world_emissions_df = world_emissions_df.pix.semijoin(models, how="left")
+
+# aggregate Serbia and Kosovo as for gfed
+country_combinations = {"srb_ksv": ["srb", "srb (kosovo)"]}
+country_emissions_df = country_emissions_df.pix.aggregate(country=country_combinations)
+
+country_emissions_df.to_csv(cams_country_temp_file)
+world_emissions_df.to_csv(cams_world_temp_file)
+
+# TODO: check! This is a tentative mapping! To which sectors should we map exactly?
+# These are the sector descriptions from eccad
+sector_mapping = {
+    "awb": "Agricultural Waste Burning",
+    "com": "Commercial",
+    "ene": "Power generation",
+    "fef": "Fugitives",
+    "ind": "Industrial process",
+    "ref": "Refineries",
+    "res": "Residential",
+    "shp": "Ships",
+    "sum": "Sum Sectors",
+    "swd": "Solid waste and waste water",
+    "tnr": "Off Road transportation",
+    "tro": "Road transportation",
+    "agl": "Agriculture livestock",
+    "ags": "Agriculture soils",
+    "fef_coal": "Fugitive coal",
+    "fef_gas": "Fugitive gas",
+    "fef_oil": "Fugitive oil",
+    "slv": "Solvents",
+    "avi": "Aviation",
+}
+
+# replace the sector mapping
+country_emissions_df = country_emissions_df.reset_index().replace({"sector": sector_mapping})
+world_emissions_df = world_emissions_df.reset_index().replace({"sector": sector_mapping})
+
+# rename to IAMC-style variable names
+country_emissions_df.insert(
+    2, "variable", "Emissions|" + country_emissions_df["gas"] + "|" + country_emissions_df["sector"]
+)
+world_emissions_df.insert(1, "variable", "Emissions|" + world_emissions_df["gas"] + "|" + world_emissions_df["sector"])
```

Comment:
In the national output, could you order the index as:

model, scenario, region, variable, unit   (change "country" to "region")

in the world output, could you do the same order, and add a new column for "region" in the same order as before, and set every entry to "World"